setup.py

```diff
@@ -1,84 +1,138 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
 import distutils.cmd
 import glob
+import json
 import os
 import pathlib
+import re
 import shutil
 import subprocess
 import sys
-import json
 import tempfile
 import urllib.request
 import zipfile
-import re
 from distutils import dir_util
 from distutils.command import build
+from distutils.dist import Distribution
 
 from setuptools import setup
 from setuptools.command import develop
 
-from azure_functions_worker import __version__
+from azure_functions_worker.version import VERSION
 
 # The GitHub repository of the Azure Functions Host
 WEBHOST_GITHUB_API = "https://api.github.com/repos/Azure/azure-functions-host"
-WEBHOST_TAG_PREFIX = "v3."
+WEBHOST_TAG_PREFIX = "v4."
 
 # Extensions necessary for non-core bindings.
 AZURE_EXTENSIONS = """\
+<?xml version="1.0" encoding="UTF-8"?>
 <Project Sdk="Microsoft.NET.Sdk">
-  <PropertyGroup>
-    <TargetFramework>netcoreapp3.1</TargetFramework>
-    <AzureFunctionsVersion>v3</AzureFunctionsVersion>
-    <WarningsAsErrors></WarningsAsErrors>
-    <DefaultItemExcludes>**</DefaultItemExcludes>
-  </PropertyGroup>
-  <ItemGroup>
-    <PackageReference
-        Include="Microsoft.NET.Sdk.Functions"
-        Version="3.0.3"
-    />
-    <PackageReference
-        Include="Microsoft.Azure.WebJobs.Extensions.CosmosDB"
-        Version="3.0.5"
-    />
-    <PackageReference
-        Include="Microsoft.Azure.WebJobs.Extensions.EventHubs"
-        Version="3.0.6"
-    />
-    <PackageReference
-        Include="Microsoft.Azure.WebJobs.Extensions.EventGrid"
-        Version="2.1.0"
-    />
-    <PackageReference
-        Include="Microsoft.Azure.WebJobs.Extensions.Storage"
-        Version="3.0.10"
-    />
-    <PackageReference
-        Include="Microsoft.Azure.WebJobs.ServiceBus"
-        Version="3.0.0-beta8"
-    />
-  </ItemGroup>
+   <PropertyGroup>
+      <TargetFramework>netcoreapp3.1</TargetFramework>
+      <AzureFunctionsVersion>v3</AzureFunctionsVersion>
+      <WarningsAsErrors />
+      <DefaultItemExcludes>**</DefaultItemExcludes>
+   </PropertyGroup>
+   <ItemGroup>
+      <PackageReference Include="Microsoft.NET.Sdk.Functions"
+        Version="4.0.1" />
+      <PackageReference Include="Microsoft.Azure.WebJobs.Extensions.CosmosDB"
+        Version="3.0.10" />
+      <PackageReference Include="Microsoft.Azure.WebJobs.Extensions.EventHubs"
+        Version="5.0.0" />
+      <PackageReference Include="Microsoft.Azure.WebJobs.Extensions.EventGrid"
+        Version="3.1.0" />
+      <PackageReference Include="Microsoft.Azure.WebJobs.Extensions.Storage"
+        Version="4.0.5" />
+      <PackageReference Include="Microsoft.Azure.WebJobs.Extensions.ServiceBus"
+        Version="4.2.1" />
+      <PackageReference
+        Include="Microsoft.Azure.WebJobs.Script.ExtensionsMetadataGenerator"
+        Version="1.1.3" />
```

Comment:
Major changes in the extension versioning. More info: https://github.com/Azure/Azure-Functions/issues/1987

Reply:
Do we need the ExtensionsMetadataGenerator? Wasn't there in older versions?

Reply:
This is now needed - I've seen some warnings, which is why I added this to remove the warnings while building.

> /home/runner/.nuget/packages/microsoft.net.sdk.functions/4.0.1/build/Microsoft.NET.Sdk.Functions.targets(74,5): warning : The ExtensionsMetadataGenerator package was not imported correctly. Are you missing '/home/runner/.nuget/packages/microsoft.azure.webjobs.script.extensionsmetadatagenerator/4.0.1/build/Microsoft.Azure.WebJobs.Script.ExtensionsMetadataGenerator.targets' or '/home/runner/.nuget/packages/microsoft.azure.webjobs.script.extensionsmetadatagenerator/4.0.1/build/Microsoft.Azure.WebJobs.Script.ExtensionsMetadataGenerator.props'? [/home/runner/work/azure-functions-python-worker/azure-functions-python-worker/build/extensions/extensions.csproj]
    1 Warning(s)